src/cmd/troubleshoot/image.go

```diff
@@ -192,83 +196,73 @@ func checkCustomModuleImagePullable(httpClient *http.Client, componentName strin
 }
 
 func imageAvailable(httpClient *http.Client, imageUrl string, apiToken string) error {
-	if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", imageUrl, "Basic", apiToken); err != nil {
+	statusCode, err := connectToDockerRegistry(httpClient, imageUrl, apiToken)
+
+	if err != nil {
 		return fmt.Errorf("registry unreachable: %w", err)
-	} else {
-		if statusCode != http.StatusOK {
-			return fmt.Errorf("image not found (status code = %d)", statusCode)
-		}
+	} else if statusCode != http.StatusOK {
```

Comment:
no need for else, when the first `if` returns

Reply:
if i had written a second if statement, there would be no need for it, as it could be concatenated with an else. I.e. what advantage does a second if-statement have to an else branch that I should change the code that way?